### PR TITLE
Fix e2e tests

### PR DESCRIPTION
**Pull request by sug0**
Very unassuming title for this issue.

Closes the following issues, hopefully:
- https://github.com/anoma/namada/issues/644
- https://github.com/anoma/namada/issues/418

### CODE REVIEW

File: tests/src/e2e/eth_bridge_tests.rs [hunk 1]
```diff
@@ -19,6 +19,8 @@ fn storage_key(path: &str) -> String {
 }
 
 #[test]
+#[ignore]
+// this test is outdated, so it is ignored
```
Comment (james-chf) [inline tests/src/e2e/eth_bridge_tests.rs:23]: Yep (also badly named...). There was this PR to clean it up (https://github.com/anoma/anoma/pull/1229) but that's a bit stale now probably. This test is still testing something we should continue testing (that a random actor cannot write arbitarily to `EthBridge` VP storage) but this test needs cleaning up.

**Comment by sug0** [comment]
We might need to port some changes from this PR in `main`: https://github.com/anoma/namada/pull/661

**Comment by sug0** [comment]
I've ran CI a couple of times, after the last fixes I pushed, and it passed every time. We could merge it as is, I guess? what do you think @james-chf @batconjurer 

image

**Review by batconjurer — APPROVED** [review]

**Comment by james-chf** [comment]
If the CI job ends up flaking a lot it could become annoying to have to keep restarting it on our PRs. If we notice any super flaky tests though we can temporarily ignore them again and open an issue, so think this PR is good to merge

**Review by james-chf — APPROVED** [review]

**Pull request merged**